File: todo-list/src/representations/Todo.js
export default class Todo {
  constructor(id, text, assignedUserId, completed = false) {
    this.id = id;
    this.text = text;
    this.assignedUserId = assignedUserId;
    this.completed = completed;
  }

  static fromState(todo) {
    return new Todo(todo.id, todo.text, todo.assignedUserId, todo.completed);
  }

  static fromApiResponse(response) {
    return new Todo(response.id, response.text, response.assignedUserId, response.completed);
  }
}

File: TodoList.Domain/Entities/Todo.cs
namespace TodoList.Domain.Entities
{
    public class Todo
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public int AssignedUserId { get; set; }
        public bool Completed { get; set; }

        public Todo()
        {}

        public Todo(int id, string text, int assignedUserId, bool completed)
        {
            Id = id;
            Text = text;
            AssignedUserId = assignedUserId;
            Completed = completed;
        }
    }
}
